apm-sniffer/apm-sdk-plugin/hutool-plugins/hutool-http-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hutool/http/v5/HutoolHttpExecuteInterceptor.java

```diff
@@ -0,0 +1,101 @@
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.hutool.http.v5;
+
+import cn.hutool.core.util.URLUtil;
+import cn.hutool.http.HttpRequest;
+import cn.hutool.http.HttpResponse;
+import java.lang.reflect.Method;
+import java.net.URL;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class HutoolHttpExecuteInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(final EnhancedInstance objInst,
+                             final Method method,
+                             final Object[] allArguments,
+                             final Class<?>[] argumentsTypes,
+                             final MethodInterceptResult result) throws Throwable {
+
+        final HttpRequest request = (HttpRequest) objInst;
+        final ContextCarrier contextCarrier = new ContextCarrier();
+        final URL url = URLUtil.url(request.getUrl());
+
+        String remotePeer = getPeer(url);
+        String formatURIPath = url.getPath();
+        AbstractSpan span = ContextManager.createExitSpan(formatURIPath, contextCarrier, remotePeer);
+
+        span.setComponent(ComponentsDefine.HUTOOL);
+        Tags.HTTP.METHOD.set(span, request.getMethod().name());
+        Tags.URL.set(span, url.toString());
+        SpanLayer.asHttp(span);
+
+        CarrierItem next = contextCarrier.items();
+        while (next.hasNext()) {
+            next = next.next();
+            request.header(next.getHeadKey(), next.getHeadValue());
+        }
+    }
+
+    @Override
+    public Object afterMethod(final EnhancedInstance objInst,
+                              final Method method,
+                              final Object[] allArguments,
+                              final Class<?>[] argumentsTypes,
+                              final Object ret) throws Throwable {
+        if (ret != null) {
+            HttpResponse response = (HttpResponse) ret;
+            int statusCode = response.getStatus();
+            if (statusCode >= 400) {
+                AbstractSpan span = ContextManager.activeSpan();
+                span.errorOccurred();
+                Tags.HTTP_RESPONSE_STATUS_CODE.set(span, statusCode);
```

Comment:
According to https://github.com/apache/skywalking-java/pull/258, the response code should be tagged no matter it is error or not.